### ISSUE TITLE
It seems to sometimes hang on a single move for over an hour

### ISSUE THREAD

**Issue by imdark** [leela-zero/leela-zero#134]
Is this reasonable?
I have 2 GTX 1080 ti, 64G of memory, i7-6950X CPU and am using the latest release on windows,
some games take under 10 minutes per game but sometimes it seems to hang for very long times, I can provide a memory dump if you want to try and review it.

this is an example of one game that seem to take very long time
Best network hash: 1ac2638de9d4d3f9eef43f2afb6b1b4c090a7c350fde4d2d77e00e39eba6fde0
Required client version: 2 (OK)
Already downloaded network.
Engine has started.
Infinite thinking time set.
1 (M9) 2 (T13) 3 (H19) 4 (K3) 5 (Q13) 6 (B19) 7 (D8) 8 (J16) 9 (T2) 10 (N13) 11 (S1) 12 (B6) 13 (O14) 14 (B4) 15 (T15) 16 (F18) 17 (K6) 18 (D6) 19 (G13) 20 (pass) 21 (B16) 22 (E19) 23 (D1) 24 (P17) 25 (E5) 26 (D7) 27 (N18) 28 (D4) 29 (G18) 30 (P12) 31 (P16) 32 (J19) 33 (F16) 34 (F9) 35 (K2) 36 (E8) 37 (L11) 38 (T12) 39 (D14) 40 (F11) 41 (H8) 42 (E3) 43 (J7) 44 (C10) 45 (F2) 46 (A3) 47 (A18) 48 (S13) 49 (Q12) 50 (pass) 51 (Q8) 52 (S15) 53 (A17) 54 (T16) 55 (B5) 56 (E10) 57 (N14) 58 (D11) 59 (D15) 60 (D13) 61 (M4) 62 (Q2) 63 (K7) 64 (S16) 65 (P15) 66 (L19) 67 (T5) 68 (P10) 69 (M7) 70 (Q19) 71 (Q1) 72 (R9) 73 (Q4) 74 (E9) 75 (J9) 76 (G7) 77 (C15) 78 (G14) 79 (M8) 80 (K15) 81 (J11) 82 (B10) 83 (P7) 84 (F15) 85 (K11) 86 (E17) 87 (S5) 88 (C13) 89 (S10) 90 (N10) 91 (P18) 92 (R10) 93 (B12) 94 (D5) 95 (Q14) 96 (Q16) 97 (H10) 98 (H3) 99 (G10) 100 (N16) 101 (G2) 102 (G4) 103 (M17) 104 (J14) 105 (E16) 106 (D10) 107 (P13) 108 (O12) 109 (Q15) 110 (R13) 111 (C9) 112 (S7) 113 (H9) 114 (R15) 115 (D9) 116 (H7) 117 (N12) 118 (P4) 119 (M13) 120 (G17) 121 (J12) 122 (B13) 123 (Q18) 124 (N4) 125 (O3) 126 (K18) 127 (G16) 128 (P2) 129 (R8) 130 (Q11) 131 (J17) 132 (O13) 133 (F12) 134 (S11) 135 (M10) 136 (E7) 137 (H6) 138 (H15) 139 (F1) 140 (B1) 141 (R19) 142 (E12) 143 (C5) 144 (B9) 145 (F10) 146 (A10) 147 (T9) 148 (O11) 149 (J6) 150 (D12) 151 (L1) 152 (K10) 153 (J3) 154 (N5) 155 (L5) 156 (T3) 157 (J18) 158 (R7) 159 (G5) 160 (M12) 161 (A8) 162 (T17) 163 (H16) 164 (H5) 165 (L15) 166 (G12) 167 (M18) 168 (R17) 169 (G6) 170 (N11) 171 (E13) 172 (F13) 173 (G3) 174 (A7) 175 (R18) 176 (E14) 177 (A16) 178 (K17) 179 (L2) 180 (L14) 181 (A11) 182 (P8) 183 (H4) 184 (O9) 185 (F4) 186 (Q17) 187 (L18) 188 (E15) 189 (G19) 190 (K4) 191 (O16) 192 (N7) 193 (R14) 194 (H17) 195 (A14) 196 (Q9) 197 (R6) 198 (K19) 199 (N17) 200 (Q7) 201 (N8) 202 (E13) 203 (M14) 204 (S8) 205 (S14) 206 (R12) 207 (H18) 208 (F19) 209 (G18) 210 (J15) 211 (O17) 212 (T14) 213 (J17) 214 (L13) 215 (N9) 216 (N15) 217 (D18) 218 (B7) 219 (O18) 220 (R8) 221 (N3) 222 (R11) 223 (L17) 224 (pass) 225 (M6) 226 (K5) 227 (P14) 228 (G8) 229 (C8) 230 (M16) 231 (F17) 232 (K8) 233 (L16) 234 (J4) 235 (G15) 236 (D16) 237 (K16) 238 (M5) 239 (F8) 240 (J18) 241 (C19) 242 (P6) 243 (P19) 244 (H1) 245 (C12) 246 (H16) 247 (L6) 248 (K12) 249 (K13) 250 (S9) 251 (T18) 252 (R5) 253 (K14) 254 (L7) 255 (F16) 256 (B11) 257 (H12) 258 (J13) 259 (J5) 260 (C7) 261 (K13) 262 (L9) 263 (P3) 264 (G16) 265 (J2) 266 (N2) 267 (R4) 268 (F7) 269 (O6) 270 (M3) 271 (J8) 272 (O7) 273 (M15) 274 (O5) 275 (T6) 276 (J17) 277 (Q5) 278 (H13) 279 (A13) 280 (E4) 281 (G11) 282 (Q6) 283 (H5) 284 (Q8) 285 (O2) 286 (G9) 287 (G4) 288 (N12) 289 (O1) 290 (P7) 291 (R5) 292 (C14) 293 (N6) 294 (E11) 295 (C16) 296 (A9) 297 (P5) 298 (T10) 299 (H11) 300 (L4) 301 (B15) 302 (K1) 303 (C11) 304 (O4) 305 (F5) 306 (M4) 307 (F3) 308 (P9) 309 (L3) 310 (T8) 311 (O8) 312 (F17) 313 (M2) 314 (K5) 315 (O4) 316 (E16) 317 (C1) 318 (F16) 319 (C4) 320 (N4) 321 (A12) 322 (M4) 323 (L12) 324 (S6) 325 (M11) 326 (N5) 327 (A4) 328 (C6) 329 (K9) 330 (E1) 331 (T4) 332 (B8) 333 (K14) 334 (A5) 335 (L14) 336 (E6) 337 (O5) 338 (E2) 339 (M3) 340 (J4) 341 (D3) 342 (A6) 343 (D2) 344 (D19) 345 (D8) 346 (S12) 347 (L4) 348 (B18) 349 (C8) 350 (A1) 351 (O15) 352 (S10) 353 (S19) 354 (M19) 355 (C17) 356 (H2) 357 (N19) 358 (C3) 359 (C5) 360 (G13) 361 (O19) 362 (M16) 363 (P1) 364 (T15) 365 (K4) 366 (C2) 367 (K5) 368 (C18) 369 (A2) 370 (N16) 371 (M5) 372 (M4) 373 (P4) 374 (N4) 375 (D9) 376 (D17) 377 (F6) 378 (M1) 379 (A19) 380 (J1) 381 (T1) 382 (R2) 383 (G1) 384 (C19) 385 (K1) 386 (H2) 387 (K12) 388 (C9) 389 (H1) 390 (E18) 391 (A15) 392 (S4) 393 (N1) 394 (S2) 395 (S3) 396 (B5) 397 (D8) 398 (P11) 399 (L13) 400 (S17) 401 (J1) 402 (J10) 403 (H18) 404 (R1) 405 (N5) 406 (B17) 407 (S4) 408 (C8) 409 (B3) 410 (M4) 411 (N4) 412 (B14) 413 (Q3) 414 (L10) 415 (A18) 416 (G15) 417 (K3) 418 (A16) 419 (C11) 420 (C1) 4

and another

Best network hash: 1ac2638de9d4d3f9eef43f2afb6b1b4c090a7c350fde4d2d77e00e39eba6fde0
Required client version: 2 (OK)
Already downloaded network.
Engine has started.
Infinite thinking time set.
1 (F16) 2 (K8) 3 (C14) 4 (C11) 5 (R11) 6 (B14) 7 (R6) 8 (N16) 9 (R1) 10 (C15) 11 (M19) 12 (M2) 13 (A12) 14 (K3) 15 (C8) 16 (O6) 17 (L3) 18 (N11) 19 (T10) 20 (A18) 21 (Q8) 22 (E18) 23 (P18) 24 (T2) 25 (O17) 26 (M4) 27 (M15) 28 (D9) 29 (S16) 30 (S3) 31 (C9) 32 (O9) 33 (J3) 34 (S2) 35 (B17) 36 (C19) 37 (R12) 38 (C17) 39 (T19) 40 (N1) 41 (Q14) 42 (L4) 43 (Q2) 44 (M7) 45 (Q1) 46 (A5) 47 (S5) 48 (C1) 49 (M14) 50 (R3) 51 (S10) 52 (H2) 53 (M6) 54 (E5) 55 (J2) 56 (K5) 57 (R19) 58 (T9) 59 (F2) 60 (G16) 61 (K19) 62 (G1) 63 (N9) 64 (D6) 65 (O19) 66 (J6) 67 (M10) 68 (L11) 69 (P6) 70 (H3) 71 (P19) 72 (G13) 73 (K4) 74 (D7) 75 (T6) 76 (J14) 77 (P5) 78 (E17) 79 (Q10) 80 (T17) 81 (D18) 82 (F11) 83 (K2) 84 (O5) 85 (N18) 86 (H5) 87 (N8) 88 (Q15) 89 (B6) 90 (M11) 91 (H9) 92 (F18) 93 (Q18) 94 (B7) 95 (G19) 96 (G18) 97 (Q3) 98 (D10) 99 (L13) 100 (K10) 101 (B12) 102 (F7) 103 (O4) 104 (E11) 105 (J12) 106 (D5) 107 (L9) 108 (E6) 109 (H14) 110 (F3) 111 (P10) 112 (B5) 113 (J17) 114 (N2) 115 (N15) 116 (J10) 117 (H8) 118 (F5) 119 (P2) 120 (M5) 121 (Q5) 122 (L19) 123 (D19) 124 (L5) 125 (K12) 126 (H13) 127 (B18) 128 (B8) 129 (N3) 130 (N4) 131 (R5) 132 (M3) 133 (A14) 134 (Q11) 135 (A17) 136 (R15) 137 (K14) 138 (G8) 139 (H4) 140 (C10) 141 (E15) 142 (F15) 143 (S13) 144 (J4) 145 (A4) 146 (A6) 147 (J9) 148 (T12) 149 (S1) 150 (B15) 151 (D8) 152 (N6) 153 (R18) 154 (K9) 155 (H6) 156 (G10) 157 (F10) 158 (F8) 159 (S11) 160 (B10) 161 (K6) 162 (O7) 163 (O16) 164 (O8) 165 (F14) 166 (K11) 167 (G5) 168 (D2) 169 (C13) 170 (M16) 171 (D12) 172 (C6) 173 (E7) 174 (J11) 175 (K15) 176 (N10) 177 (R8) 178 (A9) 179 (S4) 180 (D13) 181 (K7) 182 (D14) 183 (D1) 184 (A3) 185 (F4) 186 (C4) 187 (P7) 188 (C2) 189 (D16) 190 (L6) 191 (E16) 192 (E9) 193 (J5) 1

**Comment by qysnn** [comment]
I think it's another issue about the Windows 10 console selection mode like #71.
@gcp Maybe we should disable quick edit mode programmatically on Windows 10. I did some search and found https://stackoverflow.com/questions/13656846/how-to-programmatic-disable-c-sharp-console-applications-quick-edit-mode may be helpful.

**Comment by imdark** [comment]
@qysnn I am well aware of quick select mode, those windows are not hang due to selection, the 
"417 (K3) 418 (A16) 419 (C11) 420 (C1) 4 " is still hang after 3 hours now (i closed the other one) and it is not selected

**Comment by qysnn** [comment]
The problem about quick select mode is that it will not automatically resume even if you have cancelled your selection. Can you please check whether the program is consuming CPU and GPU resource or not when it hangs?

**Comment by imdark** [comment]
It still uses cpu and gpu about the same amount as it normally would, ~20% gpu ~5% cpu
like I said I am very aware of quick select, this happens before I clicked the window, happened 4 times so far

**Comment by qysnn** [comment]
From my point of view, the program is still running. It just no longer output to the console properly. From your output `417 (K3) 418 (A16) 419 (C11) 420 (C1) 4 `, it can be seen that the program didn't even finish the output of move 421. Even if there is something wrong in autogtp, it's unlikely to stop the output halfway. The similar thing happened to me as well before. That's why I think this may have something to do with the quick select mode even if you didn't click on the console at all. Can you try to disable quick select mode manually and restart the program to see whether it hangs?

**Comment by SebastienFaurite** [comment]
Can you try to launch it from gitbash ? I find it really more stable.

**Comment by imdark** [comment]
will try

**Comment by barrybecker4** [comment]
I have seen this behavior a couple of times today using Leela Zero 0.6 + AutoGTP v4 (on Windows 10).
If I just double click on the exe in the file browser then sometimes it plays for 2 to 5 games and then hangs - it's in the middle of a game and then just stops. I wait for 1/2 hour or so, but it never plays another move. The CPU still shows that it is 20% busy while hung.
I have also connected to Leela-zero.exe from Sabaki (as white player) and have seen similar behavior. It plays for a half dozen moves or so, and then seems to pass every move. When I click, I just place black stones.

Most recently I have tried running from both Git Bash and by double clicking on the exe. Both have been running for the last 5 hours or so. No problems with either one so far. Both have played about 33 games so far. I will let them run overnight.

**Comment by imdark** [comment]
happened again with quick select disabled
git bash is still running for now

**Comment by barrybecker4** [comment]
So I let things run overnight and both Git Bash and the cmd shell (the one started by double-clicking in exe) worked fine. However, when I turned the first 4 Editor options ("quick-edit", "insert-mode", Enable Ctrl key", "Filter clip") back on in the cmd shell (I had turned them off yestarday), and made a selection, then it stopped, and never started again. So it definitely seems like having "quick-edit" and "insert-mode" on can lead to problems in the regular shell. This is not a major issue since there is an easy work around (turn it off or use git bash instead). @imdark, did you turn off all 4 Edit Options?
